scripts/zones/Castle_Zvahl_Keep/Zone.lua

```diff
@@ -1,23 +1,55 @@
 -----------------------------------
 -- Zone: Castle_Zvahl_Keep (162)
 -----------------------------------
+local ID = zones[xi.zone.CASTLE_ZVAHL_KEEP]
+-----------------------------------
 ---@type TZone
 local zoneObject = {}
 
+local teleportTable =
+{
+    [1] = { npc = ID.npc.TELE_CENTER, event = 0, }, -- Teleports player to far NE corner
+    [2] = { npc = ID.npc.TELE_NE,     event = 2, }, -- Teleports player to far SE corner
+    [3] = { npc = ID.npc.TELE_SE,     event = 3, }, -- Teleports player to far SW corner (from middle-SE porter)
+    [4] = { npc = ID.npc.TELE_NW,     event = 1, }, -- Teleports player to far SW corner (from middle-NW porter)
+    [5] = { npc = ID.npc.TELE_SW,     event = 4, }, -- Teleports player to the top of the stairs on map 4
+    [6] = { npc = ID.npc.TELE_N,      event = 5, }, -- Teleports player to position to one of several random positions
+    [7] = { npc = ID.npc.TELE_S,      event = 7, }, -- Teleports player to position to one of several random positions
+    [8] = { npc = ID.npc.TELE_HIDDEN, event = 4, }, -- Teleports player to H-7 on map 4 platform near the ore door (south or north part randomly)
```

Comment:
Are you positive this events are the correct events relating to each trigger area?

PR #5470 was done not that long ago, captured by Teo.

From what I see, you previous work on castle Zvhal teleporters is based on ASB, and predates the changes in it.
https://github.com/AirSkyBoat/AirSkyBoat/pull/3114

Reply:
triple checking values. been doing a merry-go-round for hours now getting this right. Likely that I mixed some values up in my psychosis.

Reply:
Just did a live capture and can confirm original values are correct. Will adjust accordingly.